static/app/components/surfaceProvider.tsx

```diff
@@ -0,0 +1,16 @@
+import {createContext, type ReactNode, useContext} from 'react';
+
+export const SurfaceContext = createContext<string>('');
+
+export default function SurfaceProvider({
+  children,
+  suffix,
+}: {
+  children: ReactNode;
+  suffix: string;
+}) {
+  const parent = useContext(SurfaceContext);
```

Comment:
maybe this should be called `prefix` instead? or `base`. `parent` and `child` seem like they'd go together but `children` is used in a difference sense

Reply:
What about parentSurface?

Reply:
I like prefix too

Reply:
I think this component's gonna change a bit soon. @ryan953 mentioned surfaces where we'd want to override the current context completely, e.g. a pop-up modal unrelated to the current page.

I'm workshopping names, but it'd look something like:
`const surface = parentSurface && !override ? `${parentSurface}.${value}` : value;`
`override` is a SurfaceProvider prop and can default to false.

Reply:
`parentSurface` sounds fine to me👍